Lesson15/Ayse/script.js

```diff
@@ -0,0 +1,504 @@
+
+function checkFileExtension(filename) {
+
+  if (filename.endsWith('.pdf')) {
+    console.log('This is a PDF file.');
+  } else {
+    console.log('Not a PDF file.');
+  }
+}
+checkFileExtension('document.pdf');
+checkFileExtension('vacation.png');
+
+/*
+8. Palindrome Check
+   - Define a function `isPalindrome(str)` that checks if `str` is the same
+     forwards and backwards (normalize the string and compare it to reversed version of itself).
+   - If it is, log: "<str> is a palindrome"
+   - Otherwise, log: "<str> is not a palindrome"
+*/
+console.log("Example-8");
+
+function isPalindrome(str) {
+  const reversedStr = str.split('').reverse().join('');
+
+  if (str == reversedStr) {
+    console.log(`${str} is a palindrome`);
+  } else {
+    console.log(`${str} is not a palindrome`);
+  }
+}
+isPalindrome('level');
+isPalindrome('flower');
+
+/*
+10. Check Even or Odd (if-else)
+   - Define a function `evenOrOdd(number)` that:
+     - Logs "Even" if the number is even
+     - Logs "Odd" if the number is odd
+*/
+function evenOrOdd(number) {
+  if (number % 2 == 0) {
+    console.log(`${number}, is even.`);
+  } else {
+    console.log(`${number}, is odd.`);
+  }
+}
+evenOrOdd(5);
+evenOrOdd(10);
+
+/*
+11. URL Protocol Checker
+   - Define a function `checkProtocol(url)` that converts the URL to lowercase
+     and checks if it starts with "https" using .startsWith().
+   - Log "Secure connection" if true, otherwise "Unsecure connection".
+*/
+console.log("Example-11");
+
+function checkProtocol(url) {
+  const lowerCaseUrl = url.toLowerCase();
+  if (lowerCaseUrl.startsWith('https')) {
+    console.log('Secure connection');
+  } else {
+    console.log('Unsecure connection');
+  }
+}
+checkProtocol('https://www.google.com');
+checkProtocol('https://www.freecodecamp.org/');
+checkProtocol('//www.freecodecamp.org/');
+
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+*/
+console.log("Example-13");
+
+function repeatWord(word, times) {
+  const repeatWord = word.repeat(times);
+  console.log(repeatWord);
+}
+
+repeatWord("Hello", 3);
+repeatWord("Hi", 5);
+
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+console.log("Example-14");
+
+function censorWord(sentence, target) {
+
+  const censoredSentence = sentence.replaceAll(target, '***');
+  console.log(censoredSentence);
+}
+
+censorWord('I love apples and apples are great.', 'apples');
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+console.log("Example-15");
+
+function startsWithA(str) {
+  if (str.charAt(0) == 'A') {
```

Comment:
It could be that the given string is in lower case, so safe to add conversion of the first letter to uppercase:
```suggestion
  if (str.charAt(0).toUpperCase() === 'A') {
```